### ISSUE TITLE
Functions with generics appear to have an extra argument at runtime

### ISSUE THREAD

**Issue by Ashvin-Ranjan** [nbuilding/N-lang#287]
At runtime a function such as 
```js
let enumerate = [t] (l: list[t], default: t) -> list[(int, t)] {
    return range(0, l.len(), 1).filterMap((i: int) -> maybe[(int, t)] {
        return yes((i, l[i] | default))
    })
}
```
Should be typed as a function with two arguments, yet it is typed as 
`('infer', 'whatever', 'whatever', Tree('with_typevars', [Tree('module_type', [Token('NAME', 'list')]), Tree('tupledef', [Tree('module_type', [Token('NAME', 'int')]), Tree('module_type', [Token('NAME', 't')])])]))`

**Comment by Ashvin-Ranjan** [comment]
It appears that 
```py

            return Function(
                self,
                [self.get_name_type(arg, get_type=False) for arg in arguments],
                returntype,
                codeblock,
            )
```
Is run during runtime to get the types for the function, this causes problems however, because generic declarations are included in `arguments`, so one extra argument is passed in.

**Comment by Ashvin-Ranjan** [comment]
Fixed as of `9f75b0e`
